ddtrace/contrib/internal/grpc/client_interceptor.py

```diff
@@ -233,8 +233,8 @@ def _intercept_client_call(self, method_kind, client_call_details):
         # propagate distributed tracing headers if available
         headers = {}
         if config.grpc.distributed_tracing_enabled:
-            # NOTE: We need to pass the span to the HTTPPropagator since it isn't active at this point
-            HTTPPropagator.inject(span.context, headers, span)
+            tracer.sample(span)
```

Comment:
@mabdinur should this just be apart of the `HTTPPropagator.inject` function?

e.g. if we try to inject a context which doesn't yet have a sampling decision, make a sampling decision first?

Seems like this probably is impacting not just gRPC but any downstream propagation?

Reply:
I think this is right, based on the failing system-tests